attributes/default.rb

```diff
@@ -89,51 +99,67 @@
-                                              end
-
-when 'ubuntu18'
+if arm_instance?
+  default['cfncluster']['dcv']['supported_os'] = %w[ubuntu18 amazon2]
+  default['cfncluster']['dcv']['url_architecture_id'] = 'aarch64'
+  default['cfncluster']['dcv']['sha256sum'] = value_for_platform(
+    'centos' => {
+      '~>7' => "d590d81360b0e96652cd1ef3a74fff5cb9381f57ff55685a8ddde48d494968e4"
+    },
+    'amazon' => { '2' => "d590d81360b0e96652cd1ef3a74fff5cb9381f57ff55685a8ddde48d494968e4" },
+    'ubuntu' => { '18.04' => "0cd0512d57808cee0c48d0817a515825f9c512cb9d70e5672fecbb7450b729b3" }
+  )
+else
+  default['cfncluster']['dcv']['supported_os'] = %w[centos7 ubuntu18 amazon2]
+  default['cfncluster']['dcv']['url_architecture_id'] = 'x86_64'
+  default['cfncluster']['dcv']['sha256sum'] = value_for_platform(
+    'centos' => {
+      '~>7' => "c36020cce52ba371a796c041352db6c5d6d55d35acbbfbadafd834e1265aa5bf"
+    },
+    'amazon' => { '2' => "c36020cce52ba371a796c041352db6c5d6d55d35acbbfbadafd834e1265aa5bf" },
+    'ubuntu' => { '18.04' => "7569c95465743b512f1ab191e58ea09777353b401c1ec130ee8ea344e00f8900" }
+  )
+end
+if "#{node['platform']}#{node['platform_version'].to_i}" == 'ubuntu18'
   # Unlike the other supported OSs, the DCV package names for Ubuntu 18.04 use different architecture abbreviations than those used in the download URLs.
-  default['cfncluster']['dcv']['package_architecture_id'] = if arm_instance?
-                                                              'arm64'
-                                                            else
-                                                              'amd64'
-                                                            end
-  default['cfncluster']['dcv']['package'] = "nice-dcv-#{node['cfncluster']['dcv']['version']}-ubuntu1804-#{node['cfncluster']['dcv']['url_architecture_id']}"
-  default['cfncluster']['dcv']['server'] = "nice-dcv-server_2020.1.9012-1_#{node['cfncluster']['dcv']['package_architecture_id']}.ubuntu1804.deb" # NICE DCV server package
-  default['cfncluster']['dcv']['xdcv'] = "nice-xdcv_2020.1.338-1_#{node['cfncluster']['dcv']['package_architecture_id']}.ubuntu1804.deb"  # required to create virtual sessions
-  default['cfncluster']['dcv']['gl'] = "nice-dcv-gl_2020.1.840-1_#{node['cfncluster']['dcv']['package_architecture_id']}.ubuntu1804.deb"  # required to enable GPU sharing
-  default['cfncluster']['dcv']['sha256sum'] = if arm_instance?
-                                                "0cd0512d57808cee0c48d0817a515825f9c512cb9d70e5672fecbb7450b729b3"
-                                              else
-                                                "7569c95465743b512f1ab191e58ea09777353b401c1ec130ee8ea344e00f8900"
-                                              end
+  default['cfncluster']['dcv']['package_architecture_id'] = arm_instance? ? 'arm64' : 'amd64'
```

Comment:
why don't we move also this in the if/else above by assigning only to ubuntu 18.04 in value_for_platform?

Reply:
I decided to keep this into an if block because this variable is very specific fo ubuntu18, I agree we can do this change. We'll do in a future patch.